lib/cf/broker/src/routes/update-service.js

```diff
@@ -1,170 +1,101 @@
-    // As we build the request for pricing plan creation,
-    // 4xx errors are converted to internal server error
-    return extend({}, error, { statusCode: httpStatus.INTERNAL_SERVER_ERROR });
-
-  return error;
-};
-
-const updateMeteringPlan = (plan, planId, cb) => {
-  const meteringPlan = extend({}, plan, { plan_id: planId });
-  updatePlan('metering', meteringPlan, planId, cb);
-};
-
-const updatePricingPlan = (meteringPlan, planId, cb) => {
-  const pricingPlan = samplePricingPlan(planId,
-    config.buildPricingPlanMetrics(meteringPlan.metrics));
-
-  updatePlan('pricing', pricingPlan, planId, (error) =>
-    cb(convertBusinessError(error)));
-};
-
-const updateRatingPlan = (meteringPlan, planId, cb) => {
-  const ratingPlan = sampleRatingPlan(planId,
-    config.buildRatingPlanMetrics(meteringPlan.metrics));
-
-  updatePlan('rating', ratingPlan, planId,
-    (error) => cb(convertBusinessError(error)));
-};
+    if(!config.isServiceConfigValid(requestParameters.serviceConfig)) {
+      edebug('Service configuration is not valid, %o', requestParameters.serviceConfig);
+      response.status(httpStatus.BAD_REQUEST).send({ description: 'Invalid service configuration.' });
+      return;
+    }
 
-const updatePlans = (requestConfig, cb) => {
-  debug('Updating service instance with configuration %o', requestConfig);
-
-  const serviceConfig = requestConfig.parameters;
-  if (!serviceConfig) {
-    cb(httpStatus.OK);
-    return;
-  }
-  if(!config.isServiceConfigValid(serviceConfig)) {
-    cb(httpStatus.BAD_REQUEST, 'Invalid service configuration.');
-    return;
-  }
-
-  const instanceId = requestConfig.instance_id;
-  const planId = config.generatePlanId(instanceId, instanceId);
-  const plans = serviceConfig.plans;
-  const meteringPlan = first(plans).plan;
-
-  async.series([
-    async.apply(updateServiceMapping, requestConfig, plans, planId),
-    async.apply(updateMeteringPlan, meteringPlan, planId),
-    async.apply(updatePricingPlan, meteringPlan, planId),
-    async.apply(updateRatingPlan, meteringPlan, planId)
-  ], (err) => {
-    if (err) {
-      edebug('Failed to update service instance %o: %o', requestConfig, err);
-      const statusCode = err.statusCode ? err.statusCode : httpStatus.INTERNAL_SERVER_ERROR;
-      if (statusCode >= httpStatus.INTERNAL_SERVER_ERROR) {
-        cb(httpStatus.INTERNAL_SERVER_ERROR);
-        return;
+    const planId = config.generatePlanId(requestParameters.instance_id, requestParameters.instance_id);
+    const userProvidedPlan = getSinglePlan(requestParameters.serviceConfig);
+
+    try {
+      if (userProvidedPlan && userProvidedPlan.resource_provider) {
+        if (!config.getMappingApi()) {
+          edebug('Cannot create service mapping. No url defined.');
+          response.status(httpStatus.BAD_REQUEST).send({
+            description: 'Provided plan is not valid. Resource provider mapping is not supported.'
+            + 'Please remove "resourceProvider" field'
```

Comment:
Missing space before next sentence

Reply:
Same as above comment - we might want to rephrase this to `Invalid plan: resource provider mapping is not supported.`